### DISCUSSION TITLE
New extension type: "Global extension"

### DISCUSSION THREAD

**Discussion by Eggwise** [directus/directus#23507] [Feature Requests]
### Context

At the moment we are building an application for companies using Directus.
We are in love with the tech and the product and really appreciate all the hard work from everyone building on this amazing solution.

We are using extensions of different types like modules, interfaces, operators, all in bundles and it works like a charm.
Though recently i feel there is a need of a different kind of extension type since i want to build something that is not supporting using the current available kinds, something i think other people will also be needing and might be worth considering finding a solution for.

### Explanation & Use cases
Globally:

As a developer 
I want to have a new type of extension that makes it able to use the directus Vue API's on a more global level
So that i can build functionality that spans/lives accross all over my app.

#### Examples:

Let's say a user wants to integrate a chat in directus like Intercom, he wants to initialize the chat with the current user details and emit events to intercom on changes in the application like routing to another page.

Or lets say a user wants to add analytics to an application like using Heap.io and emit events on events in the application.

These implementations might require a component to be rendered always, (after the user logged in), not only in a specific module, layout or interface. Also these implementations should be able to access the Directus Vue API's for executing actions or fetching data.

At the moment there is (as far as i know) no extension type that supports these kind of implementations.

### Proposal

It would be nice to have a "global extension type" which developers can use to build functionality that needs to be alive allover the application, whenever the user is logged in. (or maybe even before).

@directus team

1. Am i missing something, is it already possible? If so, how do i implement it?
2. What are your thoughts on this?

### Basic Example

Examples:
Let's say a user wants to integrate a chat in directus like Intercom, he wants to initialize the chat with the current user details and emit events to intercom on changes in the application like routing to another page.

Or lets say a user wants to add analytics to an application like using Heap.io and emit events on events in the application.

These implementations might require a component to be rendered always, (after the user logged in), not only in a specific module, layout or interface. Also these implementations should be able to access the Directus Vue API's for executing actions or fetching data.

At the moment there is (as far as i know) no extension type that supports these kind of implementations.

### Motivation

A basic type of functionallity (global functionality) is hard to implement

### Detailed Design

Basicly you must be able to registe a vue component on a global level (on the page) with access to directus api's that will be alive everywhere.

### Requirements List

- Register a component that has access to directus data and api's like currentUser data that will be rendered on the page after the user is logged in

### Drawbacks

i dont know

### Alternatives

I dont know

### Adoption Strategy

They can read the docs and just add the new extension type

### Unresolved Questions

- Is it already supported?
- How will it be implemented?
- How important is it?
- How many users might benefit from this?
- How does it fit in the current roadmap?

**Comment by Eggwise** [comment]
@rijkvanzanten curious what your thoughts are on this

**Comment by JoshTheDerf** [comment]
I've gotten stuck trying to do this as well. We ultimately opted to use a hidden "module" to initialize a bunch of JS/CSS/HTML outside the Vue context and operate in parallel with the app. But it makes communication with the app inside the Vue context quite tough.
